apps/settings/js/panels/addon_details/addon_details.js

```diff
@@ -0,0 +1,161 @@
+
+    // Scroll back to the top
+    this.body.scrollTop = 0;
+
+    // Display the name of the add-on in the panel header
+    var appnameArgs = { appName: manifest.name };
+    l10n.setAttributes(this.header, 'addon-details-header', appnameArgs);
+
+    // Put an icon next to the description
+    navigator.mozApps.mgmt.getIcon(app, PREFERRED_ICON_SIZE).then((blob) => {
+      this.icon.src = URL.createObjectURL(blob);
+    }).catch(() => {
+      this.icon.src = '../style/images/default.png';
+    });
+    l10n.setAttributes(this.icon, 'accessible-app-icon', appnameArgs);
+
+    // Display the add-on description if there is one
+    if (manifest.description) {
+      // If we have a description, the ManifestHelper class returned the
+      // best localized version from the manifest file.
+      this.description.removeAttribute('data-l10n-id');
+      this.description.textContent = manifest.description;
+    } else {
+      // If there is no description, then display "no description" or similar.
+      this.description.setAttribute('data-l10n-id', 'addon-no-description');
+    }
+
+    // Display the add-on developer name if there is one
+    var developerName = manifest.developer && manifest.developer.name;
+    if (developerName) {
+      this.developer.hidden = false;
+      l10n.setAttributes(this.developer, 'addon-developer', {
+        developerName: developerName
+      });
+    } else {
+      this.developer.hidden = true;
+    }
+
+    // Get the list of targeted apps
+    this.targetsList.textContent = '';  // Clear any old content here
+    AddonManager.getAddonTargets(app).then(function(targets) {
+      var names = [];
+      targets.forEach(function(target) {
+        var manifest = new ManifestHelper(target.manifest ||
+                                          target.updateManifest);
+        names.push(manifest.name);
+      });
+
+      names.sort();
+
+      names.forEach(function(name) {
+        var item = document.createElement('li');
+        var para = document.createElement('p');
+        para.textContent = name;
+        item.appendChild(para);
+        self.targetsList.appendChild(item);
+      });
+
+      if (names.length === 0) {
+        self.targetsList.hidden = true;
+        self.noTargetsMsg.hidden = false;
+      } else {
+        self.targetsList.hidden = false;
+        self.noTargetsMsg.hidden = true;
+      }
+    });
+
+    // Hook up the enable/disable toggle button
+    this.toggle.checked = AddonManager.isEnabled(app); // set initial state
+    this.toggle.onchange = function() {                // handle user changes
+      if (self.toggle.checked) {
+        AddonManager.enableAddon(app);
+      } else {
+        AddonManager.disableAddon(app).then(function(status) {
+          // If an addon injected a script then we have to restart any
+          // affected apps. And if those apps are system apps, then we
+          // need to reboot the device.
+          if (status === 'reboot') {
```

Comment:
For these two cases, should we do anything here to make sure we do reboot / restart related apps ? Or system / Gecko would handle this for us when disabling them ?